packages/headless/src/utils/highlight.ts

```diff
@@ -11,6 +12,13 @@ export interface HighlightKeyword {
   length: number;
 }
 
+export type ResultHighlights =
```

Comment:
Hmm, it does not look right to have this type exported from headless, since it is only used inside atomic?

It would make more sense to include such a type if we exported a `highlightResult` utility, e.g.

```
highlightResult<T extends Result>(result: T, highlight: ResultHighlight, delimiters) {}
```
but I think having a dedicated function for each kind of highlight would be even more friendly

```
highlightResultExcerpt<T extends Result>(result: T, delimiters) {}
```

tl;dr I don't think we should have such a type in headless unless I am missing something?